backend/src/main/kotlin/cz/loono/backend/security/basic/CustomBasicAuthenticationEntryPoint.kt

```diff
@@ -0,0 +1,37 @@
+package cz.loono.backend.security.basic
+
+import com.google.gson.Gson
+import cz.loono.backend.api.dto.ErrorDto
+import org.apache.http.entity.ContentType
+import org.springframework.security.core.AuthenticationException
+import org.springframework.security.web.authentication.www.BasicAuthenticationEntryPoint
+import org.springframework.stereotype.Component
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+@Component
+class CustomBasicAuthenticationEntryPoint : BasicAuthenticationEntryPoint() {
+
+    override fun commence(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        authException: AuthenticationException
+    ) {
+        response.contentType = ContentType.APPLICATION_JSON.toString()
+        response.addHeader("WWW-Authenticate", "Basic realm=\"$realmName\"")
+        response.status = HttpServletResponse.SC_UNAUTHORIZED
+        response.writer.write(
+            Gson().toJson(
```

Comment:
Can't we just call `response.sendError(Int, String)` and let our global error handler return the DTO? That's why we centralized it.

Reply:
I have tried it but it's not working as expected. The global error handler doesn't catch it.